src/base.jl

```diff
@@ -460,7 +460,17 @@ function IS.add_forecast!(
             load in IS.get_components(ElectricLoad, data) if
             IS.get_uuid(get_bus(load)) in uuids
         )
-            IS.add_forecast!(data, component_, forecast, ts_data)
+            forecasts = IS.get_forecasts(component_)
+            key = IS.ForecastKey(
+                typeof(forecast),
+                get_initial_time(forecast),
+                IS.get_label(forecast),
+            )
+            if haskey(forecasts.data, key)
+                @warn "forecast $key is already stored for $(get_name(component_)), not adding $(get_name(component)) Area forecast"
+            else
+                IS.add_forecast!(data, component_, forecast, ts_data)
+            end
```

Comment:
I'm OK with this, but it seems a bit cumbersome, particularly if someone needs to do the same thing elsewhere.  As an alternative, we could add an optional flag to `IS.add_forecast` called `skip_if_present` to avoid the exception.  What do you think?

Reply:
addressed here: https://github.com/NREL-SIIP/InfrastructureSystems.jl/pull/103